Sources/TensorFlow/Layers/Core.swift

```diff
@@ -223,3 +223,26 @@ public extension Dense {
                   activation: activation)
     }
 }
+
+
+public struct Function<InputScalar: TensorFlowFloatingPoint, OutputScalar:TensorFlowFloatingPoint>: Layer {
```

Comment:
Please generalize this layer to wrap a `<T: Differentiable, U: Differentiable> @differentiable (T) -> U` function instead of wrapping a `@differentiable (Tensor<T>) -> Tensor<U>` function.

```suggestion
public struct Function<Input: Differentiable, Output: Differentiable>: Layer {
```